src/index.ts

```diff
@@ -41,7 +42,7 @@ function _withReferencePool<ModelType extends IAnyComplexType, InstanceType = In
         const id = getType(store.pool[0]).identifierAttribute || "id"
         return store.pool.find((c) => c[id] === (object as any)[id]) as InstanceType
```

Comment:
We should probably find this using [resolveIdentifier](https://mobx-state-tree.js.org/API/#resolveidentifier) instead of this loop, eventually.